src/FSharp.Core/array.fs

```diff
@@ -1578,8 +1578,7 @@ module Array =
         checkNonNull "array" array
         Microsoft.FSharp.Primitives.Basics.Array.permute indexMap array
 
-    [<CompiledName("Sum")>]
-    let inline sum (array: ^T array) : ^T =
+    let inline private classicSum (array: ^T array) : ^T =
```

Comment:
I would call it `sumImpl` to indicate it is the sum implemented here. Or maybe `fsharpSumImpl`.
Would rather avoid the term `classic`